package/machine/composition.yaml

```diff
@@ -0,0 +1 @@
+# Composition with Equinix Metal Machine / Virtual Network / Assignment
```

Comment:
@displague will this be the items we need to get an IP address for the machine? If the cluster and the machine are in the same virtual network I assume we won't need a public IP for the machine?

Reply:
The XP provisioned device will have a 10.x.x.x private address, and IPv6 public address, and an IPv4 public address by default.

We have additional options to provision devices with additional dedicated addresses or EIP addresses (or ranges):
* https://github.com/packethost/crossplane-provider-equinix-metal/blob/master/apis/server/v1alpha2/device_types.go#L176
* https://github.com/packethost/crossplane-provider-equinix-metal/blob/master/apis/server/v1alpha2/device_types.go#L141
* https://github.com/packethost/crossplane-provider-equinix-metal/blob/master/apis/server/v1alpha2/device_types.go#L163

We also have the option of adding unmanaged addresses to each node which will be available over a vlan.

---

I'm not sure which of these features are available in the cluster-api-provider.

Reply:
@displague thanks for the info! Seems like simplest solution would be to just have Machine be publicly reachable and client in the CAPI cluster pinging the server on the Machine. However, would be pretty cool for Crossplane to create the virtual network and for CAPI to deploy the cluster in it. @detiber do you know if there is support for this?

Reply:
Currently we only support the default (public) networking with cluster-api-provider-packet.

Reply:
Sounds like we will just do over public internet then.